docs/core/extensions/configuration-providers.md

```diff
@@ -0,0 +1,311 @@
+
+- Are only set in processes launched from the command window they were set in.
+- Won't be read by web apps launched with Visual Studio.
+
+The following [setx](/windows-server/administration/windows-commands/setx) commands can be used to set the environment keys and values on Windows. Unlike `set`, `setx` settings are persisted. `/M` sets the variable in the system environment. If the `/M` switch isn't used, a user environment variable is set.
+
+```dotnetcli
+setx SecretKey "Secret key from setx environment" /M
+setx TransientFaultHandlingOptions__Enabled "true" /M
+setx TransientFaultHandlingOptions__AutoRetryDelay "00:00:05" /M
+
+dotnet run
+```
+
+To test that the preceding commands override *appsettings.json* and *appsettings.*`Environment`*.json*:
+
+- With Visual Studio: Exit and restart Visual Studio.
+- With the CLI: Start a new command window and enter `dotnet run`.
+
+Call <xref:Microsoft.Extensions.Configuration.EnvironmentVariablesExtensions.AddEnvironmentVariables%2A> with a string to specify a prefix for environment variables:
+
+:::code language="csharp" source="snippets/configuration/console-env/Program.cs" highlight="15-16":::
+
+In the preceding code:
+
+- `config.AddEnvironmentVariables(prefix: "CustomPrefix_")` is added after the default configuration providers. For an example of ordering the configuration providers, see [XML configuration provider](#xml-configuration-provider).
+- Environment variables set with the `CustomPrefix_` prefix override the default configuration providers. This includes environment variables without the prefix.
+
+The prefix is stripped off when the configuration key-value pairs are read.
+
+The following commands test the custom prefix:
+
+```dotnetcli
+set CustomPrefix__SecretKey="Secret key with CustomPrefix_ environment"
+set CustomPrefix_TransientFaultHandlingOptions__Enabled=true
+set CustomPrefix_TransientFaultHandlingOptions__AutoRetryDelay=00:00:21
+
+dotnet run
+```
+
+The default configuration loads environment variables and command-line arguments prefixed with `DOTNET_`. The `DOTNET_` prefix is used by .NET for host and app configuration, but not for user configuration.
+<!-- For more information on host and app configuration, see .NET Generic Host. -->
+
+On [Azure App Service](https://azure.microsoft.com/services/app-service), select **New application setting** on the **Settings > Configuration** page. Azure App Service application settings are:
+
+- Encrypted at rest and transmitted over an encrypted channel.
+- Exposed as environment variables.
+
+### Connection string prefixes
+
+The Configuration API has special processing rules for four connection string environment variables. These connection strings are involved in configuring Azure connection strings for the app environment. Environment variables with the prefixes shown in the table are loaded into the app with the [default configuration](#default) or when no prefix is supplied to `AddEnvironmentVariables`.
```

Comment:
This line introduced a build warning @IEvangelist .

Reply:
I'll fix that, thanks